test.js

```diff
@@ -134,6 +134,50 @@ test('execa.shellSync() includes stdout and stderr in errors for improved debugg
 	t.throws(() => execa.shellSync('node fixtures/error-message.js'), {message: STDERR_STDOUT_REGEXP, code: 1});
 });
 
+test.cb('execa() with .kill after it with SIGKILL should kill cleanly', t => {
+	const proc = execa('node', ['fixtures/sub-process-no-killable']);
+	t.is(typeof proc.pid, 'number');
+	setTimeout(() => {
+		proc.kill('SIGKILL');
+	}, 100);
+
+	setTimeout(() => {
+		t.false(isRunning(proc.pid));
+		t.end();
+	}, 2000);
+});
+
+test.cb('execa() with .kill after it with SIGTERM should not kill (no retry)', t => {
```

Comment:
The ideas behind the two tests look good, but with the same comments as the first test.

Reply:
Done it but the second test is not upposed to die but dies anyway. This is a bit weird to me. You have an idea?

Reply:
I just tried it by cloning your work, and for me the child process does not die. The test passes. Do you get different results?

Reply:
Are you sure it's running? When I add `t.true(isRunning(subprocess.pid));` I get not passing tests. Since the fixture eats all `SIGTERM`s, the process should not finish.

Reply:
Yes you're right it dies but should not.

I figured out the problem: it's a race condition. The fixture file starts running after  `subprocess.kill()` is executed. I.e. when it receives the `SIGTERM` signal, it has not setup any signal handler yet. I.e. it is killed.

So you need to ensure the `SIGTERM` signal is fired after the fixture file has loaded. Possible solutions:
  - `await promisify(setTimeout)(1e3)` before sending `SIGTERM`. Now that one feels a little hacky.
  - use `ipc` and make the fixture file call `process.send()` after setting up `SIGTERM` handler to communicate "I am ready, please send me your signal".
  - is there another way to know that a child process has finished loading? I am not sure.

You would actually need to do that in all tests, not just that test.

How I debugged this problem: I added a `assert.fail()` at the beginning of the fixture file. Since `await subprocess` was throwing with `SIGTERM`, it meant it did not even reach that `assert.fail()`.

Reply:
Amazing, thanks for bearing with me and give me so much guidance.
I added an ipc listener, now all tests work as expected: c0c3527d06777617a30b6e4391442f5a5eef3244